tests/by-util/test_tail.rs

```diff
@@ -2512,722 +2590,721 @@ fn test_illegal_seek() {
+#[test]
+fn test_pipe_when_lines_option_given_input_size_is_one_byte_greater_than_buffer_size() {
+    let total_lines = 100;
+    let random_string = RandomString::generate_with_delimiter(
+        Alphanumeric,
+        b'\n',
+        total_lines,
+        true,
+        CHUNK_BUFFER_SIZE + 1,
+    );
+    let random_string = random_string.as_str();
+    let lines = random_string.split_inclusive('\n');
 
-        let expected = lines.clone().skip(total_lines - 1).collect::<String>();
-        new_ucmd!()
-            .args(&["-n", "-1"])
-            .pipe_in(random_string)
-            .ignore_stdin_write_error()
-            .succeeds()
-            .stdout_only(expected);
+    new_ucmd!()
+        .args(&["-n", "+0"])
+        .pipe_in(random_string)
+        .ignore_stdin_write_error()
+        .succeeds()
+        .stdout_only(random_string);
 
-        let expected = lines.clone().skip(1).collect::<String>();
-        new_ucmd!()
-            .args(&["-n", "+2"])
-            .pipe_in(random_string)
-            .ignore_stdin_write_error()
-            .succeeds()
-            .stdout_only(expected);
+    let expected = lines.clone().skip(total_lines - 1).collect::<String>();
+    new_ucmd!()
+        .args(&["-n", "-1"])
+        .pipe_in(random_string)
+        .ignore_stdin_write_error()
+        .succeeds()
+        .stdout_only(expected);
 
-        let expected = lines.clone().skip(1).collect::<String>();
-        new_ucmd!()
-            .args(&["-n", "-99"])
-            .pipe_in(random_string)
-            .ignore_stdin_write_error()
-            .succeeds()
-            .stdout_only(expected);
-    }
+    let expected = lines.clone().skip(1).collect::<String>();
+    new_ucmd!()
+        .args(&["-n", "+2"])
+        .pipe_in(random_string)
+        .ignore_stdin_write_error()
+        .succeeds()
+        .stdout_only(expected);
 
-    #[test]
-    fn test_pipe_when_lines_option_given_input_size_has_multiple_size_of_buffer_size() {
-        let total_lines = 100;
-        let random_string = RandomString::generate_with_delimiter(
-            Alphanumeric,
-            b'\n',
-            total_lines,
-            true,
-            CHUNK_BUFFER_SIZE * 3 + 1,
-        );
-        let random_string = random_string.as_str();
-        let lines = random_string.split_inclusive('\n');
+    let expected = lines.clone().skip(1).collect::<String>();
+    new_ucmd!()
+        .args(&["-n", "-99"])
+        .pipe_in(random_string)
+        .ignore_stdin_write_error()
+        .succeeds()
+        .stdout_only(expected);
+}
 
-        new_ucmd!()
-            .args(&["-n", "+0"])
-            .pipe_in(random_string)
-            .ignore_stdin_write_error()
-            .succeeds()
-            .stdout_only(random_string);
+// FIXME: windows: this test failed with timeout in the CI. Running this test in
+// a Windows VirtualBox image produces no errors.
+#[test]
+#[cfg(not(target_os = "windows"))]
```

Comment:
In case a test runs locally but not in the CI, I would use `is_ci()`, instead of disabling the test completely.
e.g.
```rust
if cfg!(windows) && is_ci() {
    println!("TEST SKIPPED (<reason>)");
    return;
}
```

Reply:
won't fix

Reply:
Please explain why "Won't fix"

Reply:
I won't change it, because it's correct the way it is. It clearly indicates the implementation has a problem on windows. The comment just explains what I've found out and is a help for others. Sure it's failing in the ci and passing on my pc and my virtualbox but that doesn't mean there's something wrong with the implementation for windows, so it is switched off entirely. Windows pipes are different from unix pipes and fixing this in this pr is out of the scope of this pr. Additionally, if someone's running the test suite on windows but working on uutils in a different place he may probably run in this timeout.

Reply:
@jhscheer ok with you?

Reply:
Yes.